### PR TITLE
Adding SIKE scheme, implementations and benchmarks

### PR DESCRIPTION
**Pull request by taudor**

### CODE REVIEW

**Comment by dingens** [comment]
`bench/optX64assp503compressed_p503_compressed_skylake.yaml` (et al) should not contain the second underscore (because the parameter set is named `p503compressed`, not `p503_compressed`; and this should not validate actually, am I right?

**Comment by taudor** [comment]
The problem is, that the implementation is called `optX64assp503compressed` and the parameter set is called `p503compressed`. SIKE has for almost all implementations different sizes. That's why I included them all as individual implementations.

It looks really confusing, do you have maybe a suggestion how to solve this?

**Comment by dingens** [comment]
You mean the duplication? That's unfortunate but not big of a deal (nobody will ever see the long string in practice).

But I was refering just to the detail that it is called
`bench/optX64assp503compressed_`**`p503_compressed`**`_skylake.yaml`
but should be called 
`bench/optX64assp503compressed_`**`p503compressed`**`_skylake.yaml`
- the boldened part is the parameter set, which doesn't contain an underscore. This should be fixed here; and additionally we need to review the validation because it didn't detect this.

**Comment by taudor** [comment]
Now I get what you mean, yes the second underscore should not be there.

File: encryption/sike/sike-shake256/impl/optX64assp434.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: Optimized implementation using x64 assembly for p434
+type: optimized
+platform: c, assembly
```
[the remark this answers is not in the document]
Comment (taudor) [reply encryption/sike/sike-shake256/impl/optX64assp434.yaml:3]: True, I'll add that.

File: encryption/sike/sike-shake256/impl/optX64assp434.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+side channel guarding:
+  branching: true
+  timing: true
```
[the remark this answers is not in the document]
Comment (taudor) [reply encryption/sike/sike-shake256/impl/optX64assp434.yaml:6]: Do we have this kind of information for enough other schemes? Could be interesting information.

File: encryption/sike/sike-shake256/sike-shake256.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+name: SIKE-SHAKE256
+type: KEM
+security notion: IND-CCA
```
[the remark this answers is not in the document]
Comment (taudor) [reply encryption/sike/sike-shake256/sike-shake256.yaml:3]: I'll add something.

File: encryption/sike/sike-shake256/impl/optX64assp434.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dingens) [reply encryption/sike/sike-shake256/impl/optX64assp434.yaml:6]: If we add it to the schema, it will incentivize people to add this info, won't it?

**Pull request merged**